src/main/java/com/greghaskins/spectrum/dsl/specification/Specification.java

```diff
@@ -31,11 +34,7 @@
    *                define each expected behavior
    */
   static void describe(final String context, final Block block) {
-    final Suite suite = DeclarationState.instance()
-        .getCurrentSuiteBeingDeclared()
-        .addSuite(context);
-    suite.applyPreconditions(block);
-    DeclarationState.instance().beginDeclaration(suite, block);
+    addSuiteInternal(parent -> parent.addSuite(context), block);
```

Comment:
There some stuff about creating a suite that was common to more than one implementation - extracted out to a common helper.